src/extensions/surveys.tsx

```diff
@@ -77,6 +77,10 @@ export class SurveyManager {
                 return
             }
         }
+        // Adding this duplicate check to see if there's some sort of race-condition between getActiveMatchingSurveys and actual URL changes, as sometimes Surveys are shown up when they shouldn't be
+        if (!doesSurveyUrlMatch(survey)) {
+            return
+        }
```

Comment:
should that be in the `callSurveysAndEvaluateDisplayLogic` method then so it'd hopefully fix for handleWidgetSelector as well?

Reply:
yeah that makes more sense - just moved now